JamJar/VideoPageViewController.swift

```diff
@@ -94,10 +110,142 @@ class VideoPageViewController: BaseViewController, UITableViewDelegate, UITableV
+                    self.updateLikeDislikeButtons()
+                }
+            }
+        } else {
+            VideoService.vote(video.id, vote: true) { success, error in
+                if !success {
+                    // Error - show the user
+                    let errorTitle = "Video error!"
+                    if let error = error { SCLAlertView().showError(errorTitle, subTitle: error, closeButtonTitle: "Got it") }
+                    else { SCLAlertView().showError(errorTitle, subTitle: "", closeButtonTitle: "Got it") }
+                } else {
+                    self.video.userVote = true
+                    self.updateLikeDislikeButtons()
+                }
+            }
+        }
+    }
+    
+    @IBAction func dislikePressed(sender: UIButton) {
+        if(video.userVote == false) {
+            VideoService.vote(video.id, vote: nil) { success, error in
+                if !success {
+                    // Error - show the user
+                    let errorTitle = "Video error!"
+                    if let error = error { SCLAlertView().showError(errorTitle, subTitle: error, closeButtonTitle: "Got it") }
+                    else { SCLAlertView().showError(errorTitle, subTitle: "", closeButtonTitle: "Got it") }
+                } else {
+                    self.video.userVote = nil
+                    self.updateLikeDislikeButtons()
+                }
+            }
+        } else {
+            VideoService.vote(video.id, vote: false) { success, error in
+                if !success {
+                    // Error - show the user
+                    let errorTitle = "Video error!"
+                    if let error = error { SCLAlertView().showError(errorTitle, subTitle: error, closeButtonTitle: "Got it") }
+                    else { SCLAlertView().showError(errorTitle, subTitle: "", closeButtonTitle: "Got it") }
+                } else {
+                    self.video.userVote = false
+                    self.updateLikeDislikeButtons()
+                }
+            }
+        }
+    }
+    
+    @IBAction func flagVideoTapped(sender: UIButton) {
+        let flagView = SCLAlertView()
+        
+        // Creat the subview
+        let subview = UIView(frame: CGRectMake(0,0,216,100))
+        let x = (subview.frame.width - 180) / 2
+        
+        // reason label
+        let reasonLabel = UILabel(frame: CGRectMake(x,5,180,25))
+        reasonLabel.text = "Reason:"
+        subview.addSubview(reasonLabel)
+        
+        // reason PickerView
+        let reasonPickerView = UIPickerView(frame: CGRectMake(x,10,180,100))
+        reasonPickerView.delegate = self
+        reasonPickerView.dataSource = self
+        subview.addSubview(reasonPickerView)
+        
+        // Add the subview to the alert's UI property
+        flagView.customSubview = subview
+        
+        // Add Notes Text Field
+        let notes = flagView.addTextField("Notes")
+        
+        flagView.addButton("Submit") {
```

Comment:
More cloning than Dolly :sheep: :sheep: :sheep: :sheep: (http://www.animalresearch.info/en/medical-advances/timeline/cloning-dolly-the-sheep/)